### ISSUE TITLE
dng file from mobile phone Nexus 5x can't be edited "failed to read camera white balance information"

### ISSUE THREAD

**Issue by michakrause** [darktable-org/darktable#2498]
**Describe the bug**
Im trying to edit raw files from a Mobile Phone (Nexus 5X), files show thumbnails in lighttable, but opening them in darkroom gives errors:
"failed to read camera white balance information from `filename.dng'!"
"image `filename.dng' is not available!"

**To Reproduce**
open attached raw file in darkroom

**Expected behavior**
File can be opened

**Platform (please complete the following information):**
 - OS: Debian 9.8
 - Version 2.6.0
[IMG_20190426_123914_484.dng.zip](https://github.com/darktable-org/darktable/files/3134996/IMG_20190426_123914_484.dng.zip)

**Comment by LebedevRI** [comment]
That is not a DNG/TIFF.
```
$ xxd -l 16 /tmp/IMG_20190426_123914_484.dng
00000000: ffd8 ffe0 0010 4a46 4946 0001 0100 0001  ......JFIF......
```
First two bytes are `FF` `D8`, they should be either "MM" (`4D``4D`) or "II" (`49``49`).
You should report that to whatever means (the capture software) you used
to produce that image, they are clearly producing bogus files.

```
$ file /tmp/IMG_20190426_123914_484.dng
/tmp/IMG_20190426_123914_484.dng: JPEG image data, JFIF standard 1.01, aspect ratio, density 1x1, segment length 16, Exif Standard: [TIFF image data, little-endian, direntries=5], baseline, precision 8, 4016x3008, components 3
```

**Comment by LebedevRI** [comment]
Moreover, that _appears_ to be a traditional 8-bit jpeg, not a raw.

**Comment by pphotography** [comment]
Same for me. When I open the DNG in Irfanview, the program tells me that it is a JPG with incorrect file extension.
I open the file in a text editor, the first bytes say `ÿØÿà JFIF ` which is the beginning of a JPG.

**Comment by LebedevRI** [comment]
That magic (`ff` `d8`) is already marked as LDR:
https://github.com/darktable-org/darktable/blob/8e401e7f098d9e4cbd7852e95cf2a224326a411e/src/common/imageio.c#L370-L371
It is already checked first, before going to rawspeed:
https://github.com/darktable-org/darktable/blob/8e401e7f098d9e4cbd7852e95cf2a224326a411e/src/common/imageio.c#L947-L948
But `dt_imageio_open_jpeg()` (which is the one that succeeds in loading it) will only load if you rename it to `.JPG`:
https://github.com/darktable-org/darktable/blob/8e401e7f098d9e4cbd7852e95cf2a224326a411e/src/common/imageio_jpeg.c#L731-L732

So TLDR: first complain to the whatever origin software that produce those files, and if you still need them, rename them `s/.dng/.jpg/`.

**Comment by michakrause** [comment]
Hm, I don't think it's just a jpeg, because I can open the file in Lightroom mobile, and there it behaves like a raw file, meaning I can change the white balance without any noticeable loss in quality, and can recover highlights a little.

**Comment by parafin** [comment]
Are you sure the file you provided wasn't modified by other software? There are mentions of Picasa and Photoshop in EXIF data. Also for 12MP picture 1.5MB filesize is an obvious sign that it's plain JPEG, not RAW. I don't think that even lossy compressed RAW can be that small.

**Comment by michakrause** [comment]
@parafin you are right. I took a new picture, and it is 24MB in size. I think the file was indeed modified probably by up and downloading it through Google-Photos. I did not think that they would change raw/dng files.
If I open the 24MB file in Darktable I can edit it with no problems.
Thank you for looking into it, sorry for the trouble.
